README.md

```diff
@@ -1,25 +1,25 @@
 # VML
 
 This package provides bindings to the Intel Vector Math Library for
-arithmetic and transcendental functions. It is often substantially
-faster than using Julia's built-in functions.
+arithmetic and transcendental functions. Especially for large vectors it is often substantially faster than broadcasting Julia's built-in functions.
 
-## Using VML.jl
+## Setting up VML.jl
 
 To use VML.jl, you must have the Intel Vector Math Library installed.
-This is included in [MKL](http://software.intel.com/en-us/intel-mkl),
-which is free for non-commercial use. You must also copy/symlink the
-appropriate shared library to a directory known to the linker (e.g.
-`/usr/local/lib`) or you must modify the path to `lib` in `src/VML.jl`.
+For this you have two options. First, you can install (and build) [MKL.jl](https://github.com/JuliaComputing/MKL.jl), which will add the necessary libraries to your Julia install. This will change your Julia system image however, so if you would prefer not to do that you can get the stand-alone [MKL](http://software.intel.com/en-us/intel-mkl),
+which is free for non-commercial use. 
+
+Libdl automatically finds the relevant libraries in the default search locations. This already the case if you use MKL.jl, but the stand-alone may require you to source `mklvars.sh`. The default command on Mac and Ubuntu is `source /opt/intel/mkl/bin/mklvars.sh intel64`. You may want to add this to your `.bashrc`. 
+Adding a new `*.conf` file in `/etc/ld.so.conf.d` also works, as the `intel-mkl-slim` package in the AUR does automatically. 
+
+Using [CpuId.jl](https://github.com/m-j-w/CpuId.jl), VML.jl detects if your processor supports the newer `avx2` instructions, and if not default to `libmkl_vml_avx`. If your system does not have AVX this package will currently not work for you.
+
+If the CPU feature detection does not work for you, please open an issue. 
 
-Currently, VML.jl is configured to use `libmkl_vml_avx`, which requires
-AVX support. If your system does not have AVX (e.g., most pre-Sandy
-Bridge systems), you will need to modify the `const lib` declaration at
-the top of `src/VML.jl`. Future versions of VML.jl may automatically
-detect CPU architecture.
+## Using VML.jl
 
 After loading VML.jl, vector calls to functions listed below will
-automatically use VML instead of openlibm when possible.
+automatically use VML instead of openlibm when possible. Note that most function currently do not have a vectorized version (e.g. you call `sin.(rand(300))`), so there should be no conflict. If there is let me know. Updated and conflict tested exported functions are planned for the future.
```

Comment:
There will be conflicts for `exp` and `log`, which act on matrices in Base, as do many trig functions. 
```
julia> exp([1 1; 1 1.0])
2×2 Array{Float64,2}:
 4.19453  3.19453
 3.19453  4.19453

julia> VML.exp([1 1; 1 1.0])
2×2 Array{Float64,2}:
 2.71828  2.71828
 2.71828  2.71828
```

Reply:
Added the example to the readme